MatrixKit/Models/Room/MXKAttachment.m

```diff
@@ -163,13 +185,109 @@ - (NSURL *)getThumbnailUrlForSize:(CGSize)size
     return nil;
 }
 
-- (void)getThumbnail:(void (^)(UIImage *))onSuccess failure:(void (^)(NSError *error))onFailure {
+- (NSString *)thumbnailMimeType
+{
     NSDictionary *thumbnail_file = self.event.content[@"thumbnail_file"];
-    if (thumbnail_file && thumbnail_file[@"url"]) {
+    if (thumbnail_file && thumbnail_file[@"mimetype"])
+    {
+        return thumbnail_file[@"mimetype"];
     }
+    return nil;
+}
+
+- (UIImage *)getCachedThumbnail {
+    NSString *cacheFilePath = [MXKMediaManager cachePathForMediaWithURL:self.thumbnailURL
+                                                                andType:self.thumbnailMimeType
+                                                               inFolder:kMXKMediaManagerDefaultCacheFolder];
```

Comment:
Attachment files should be stored in their room folder so that we delete them when the user leaved the room.